frontend/src/pages/Traces/useGetTraces.ts

```diff
@@ -153,74 +133,42 @@ export const useGetTraces = ({
 		}, []),
 	})
 
-	const fetchMoreForward = useCallback(async () => {
-		if (!windowInfo.hasNextPage || loadingAfter) {
-			return
-		}
-
-		const lastItem = data && data.traces.edges[data.traces.edges.length - 1]
-		const lastCursor = lastItem?.cursor
-
-		if (!lastCursor) {
-			return
-		}
-
-		setLoadingAfter(true)
-
-		await fetchMore({
-			variables: { after: lastCursor },
-			updateQuery: (prevResult, { fetchMoreResult }) => {
-				const newData = fetchMoreResult.traces.edges
-				setWindowInfo({
-					...windowInfo,
-					hasNextPage: fetchMoreResult.traces.pageInfo.hasNextPage,
-				})
-				setLoadingAfter(false)
-				return {
-					traces: {
-						...prevResult.traces,
-						edges: [...prevResult.traces.edges, ...newData],
-						pageInfo: fetchMoreResult.traces.pageInfo,
-					},
-				}
-			},
-		})
-	}, [data, fetchMore, loadingAfter, windowInfo])
-
-	const fetchMoreBackward = useCallback(async () => {
-		if (!windowInfo.hasPreviousPage || loadingBefore) {
-			return
-		}
-
-		const firstItem = data && data.traces.edges[0]
-		const firstCursor = firstItem?.cursor
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	const fetchMoreForward = useCallback(
+		debounce(async () => {
```

Comment:
Noticed a delay here after hitting the bottom of the page, I think because we wait for the 500ms before executing the function. What do you think about setting `leading: true` in the options so it is invoked immediately when hit the first time?

https://lodash.com/docs/4.17.15#debounce

Reply:
Good catch! Lets do that

Reply:
Just for history, had to break out the request into its own callback to avoid the function being recreated when the state changes, resulting in a race condition and duplicate request sent